packages/errors/src/index.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Creates a new error type with the given name and parent class. Sets up
+ * boilerplate necessary to successfully subclass Error and preserve stack trace
+ * @param name - The name of the error type
+ * @param ParentType - The error that serves as the parent
+ * @returns The new type
+ */
+export const createErrorType = (
+  name: string,
+  ParentType: new (message: string) => Error = Error
+) =>
+  (
+    class extends ParentType {
+      constructor(message: string) {
+        super(message);
+        this.name = name;
+        Error.captureStackTrace(this, this.constructor);
+      }
+    }
+  );
+
+interface ErrorWithOptionalCode extends Error {
+  code?: string;
+}
+
+/**
+ * Test to see if a given exception is an AWS Throttling Exception
+ */
+export const isThrottlingException = (err: ErrorWithOptionalCode) =>
+  err.code === 'ThrottlingException';
+
+/**
+ * Returns true if the error is a resource error.
+ */
+export const isWorkflowError = (error: Error) => error.name.includes('WorkflowError');
+
+/**
+ * Returns true if the error is a DynamoDB conditional check exception.
+ */
+export const isConditionalCheckException = (error: { code?: string }) =>
```

Comment:
Use the `ErrorWithOptionalCode` interface here

Reply:
Fixed, good catch.